open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -460,15 +460,16 @@ def __get_sms_object(self, hackathon_id):
         :return: ServiceManagementService object
         :rtype: class 'azure.servicemanagement.servicemanagementservice.ServiceManagementService'
         """
-        hackathon_azure_key = self.db.find_first_object_by(HackathonAzureKey, hackathon_id=hackathon_id)
+        hackathon_azure_keys = Hackathon.objects(id=hackathon_id).first().azure_keys
 
-        if hackathon_azure_key is None:
+        if hackathon_azure_keys is None:
```

Comment:
`hackathon_azure_keys ` is of type `list` should never be `None` since it's default value is `[]` which is not None. Use `len(hackathon_azure_keys) == 0` instead